File: arrays/reverse_array_string_in_place/reverse_array_string_in_place.py
# Reverse a given list of characters in place
def reverse(string):
  first_idx = 0
  last_idx = len(string) - 1

  while first_idx < last_idx:
    latest_char = string[last_idx]
    string[last_idx] = string[first_idx]
    string[first_idx] = latest_char
    first_idx += 1
    last_idx -= 1

  
arr = ["l", "a", "t", "t", "e"]
reverse(arr)
print(arr)
  

File: arrays/reverse_array_string_in_place/reverse_array_string_in_place.js
// Reverse a given array of characters in place 
function reverse(stringArr) {
  let firstIdx = 0;
  let lastIdx = stringArr.length - 1;

  while (firstIdx < lastIdx) {
    const latestChar = stringArr[lastIdx];
    stringArr[lastIdx] = stringArr[firstIdx];
    stringArr[firstIdx] = latestChar;
    firstIdx++;
    lastIdx--;
  }
}

let arr = [1,2,3,4,5];
console.log(arr);
reverse(arr);
console.log(arr);